code/frameworks/nextjs/README.md

```diff
@@ -167,22 +170,137 @@ export default function Home() {
 }
 ```
 
-#### Optimization
+#### AVIF
 
-All Next.js `Image`s are automatically [unoptimized](https://nextjs.org/docs/api-reference/next/image#unoptimized) for you.
+This format is not supported by this framework yet. Feel free to [open up an issue](https://github.com/storybookjs/storybook/issues) if this is something you want to see.
 
-If [placeholder="blur"](https://nextjs.org/docs/api-reference/next/image#placeholder) is used, the [blurDataURL](https://nextjs.org/docs/api-reference/next/image#blurdataurl) used is the [src](https://nextjs.org/docs/api-reference/next/image#src) of the image (thus effectively disabling the placeholder).
+### Next.js Navigation
 
-See [this issue](https://github.com/vercel/next.js/issues/18393) for more discussion on how Next.js `Image`s are handled for Storybook.
+Please consider, that [next/navigation](https://beta.nextjs.org/docs/upgrade-guide#step-5-migrating-routing-hooks) can only be used in components/pages of the `app` directory of Next.js v13 or higher.
 
-#### AVIF
+#### Set `nextAppDirectory` to `app`
 
-This format is not supported by this framework yet. Feel free to [open up an issue](https://github.com/storybookjs/storybook/issues) if this is something you want to see.
+If your story imports components that use `next/navigation`, you need to set the parameter `nextAppDirectory` to `true` in your Story:
+
+```js
+// SomeComponentThatUsesTheRouter.stories.js
+import SomeComponentThatUsesTheNavigation from './SomeComponentThatUsesTheNavigation';
+
+export default {
+  component: SomeComponentThatUsesTheNavigation,
+};
+
+// if you have the actions addon
+// you can click the links and see the route change events there
+export const Example = {
+  parameters: {
+    nextAppDirectory: true,
```

Comment:
WDYT about scoping this:

```
parameters: {
  nextjs: {
    appDirectory: true,
  }
}
```

This way we namespace the parameters and they will be easier to add TS types later

Reply:
One of the parameters we copied from `storybook-addon-next` is `nextRouter`. We left it as-is to provide a smooth upgrade experience for users of that addon.

If we don't think that's important, I agree that this is better:

```
parameters: {
  nextjs: {
    appDirectory: ...,
    router: ...,
  }
}
```

Reply:
This would be a breaking change, especially if we want to move `parameters.nextRouter` to `parameters.nextjs.nextRouter`. What do you think? Is that ok for Storybook 7 to introduce that breaking change?

Reply:
Fixed. Take a look: https://github.com/storybookjs/storybook/pull/20065/commits/1dbaac55216821e920e07f7e87d7872e9a9a7ea3